### ISSUE TITLE
Questions About the feature_visualization Function

### ISSUE THREAD

**Issue by Zengyf-CVer** [ultralytics/yolov5#3914]
@glenn-jocher 

> This is my first question:

I tried the feature_visualization function and found some problems.
First, when I run this detection code, two directories are generated in `runs/features`: `exp` and `exp2`, as shown in the figure:
```shell
python detect.py --weights ./weight/yolov5s.pt --source ./train2017/000000000081.jpg
```
![ksnip_20210707-100539](https://user-images.githubusercontent.com/41098760/124689850-da989880-df0b-11eb-9914-0ccbd2cf1d86.png)
Display in the shell:
![ksnip_20210707-102925](https://user-images.githubusercontent.com/41098760/124691428-657a9280-df0e-11eb-8e2e-18e385e1e6f7.png)

The picture above is the first saved file, in `exp`. 
In the display of the shell, there is another one that processes the input image, as shown in the figure:
![ksnip_20210707-100511](https://user-images.githubusercontent.com/41098760/124691459-6e6b6400-df0e-11eb-9ef8-200c54a76a11.png)

So I checked the specific feature map generated:
In `exp`:
![stage_8_SPP_features](https://user-images.githubusercontent.com/41098760/124690602-07997b00-df0d-11eb-8686-b8cb545af380.png)
In `exp2`:
![stage_8_SPP_features](https://user-images.githubusercontent.com/41098760/124690631-13853d00-df0d-11eb-9b3d-a2ed58035428.png)
**So what I want to ask is, why are two directories `exp` and `exp2` generated at the same time? What do these two directories represent?**

> This is my second question:

I checked the feature_visualization function in plots.py, there are two valuable codes:https://github.com/ultralytics/yolov5/blob/33202b7f0bbfcc55860e8c0cafbf13f227e77a84/utils/plots.py#L465 and
https://github.com/ultralytics/yolov5/blob/33202b7f0bbfcc55860e8c0cafbf13f227e77a84/utils/plots.py#L468

**I am not sure, whether the generated two directories `exp` and `exp2` are related to the models type I set?**
https://github.com/ultralytics/yolov5/blob/33202b7f0bbfcc55860e8c0cafbf13f227e77a84/models/yolo.py#L158-L159

**What else are settings like `models.common.SPP`?**
**I am not sure about the specific architecture of yolov5. Is there any relevant architecture diagram? Like this, but the official version?**

![image](https://user-images.githubusercontent.com/41098760/124692886-fa7e8b00-df10-11eb-8163-6e17711e5897.png)

**Looking forward to your reply, thank you very much.**

**Comment by glenn-jocher** [comment]
@Zengyf-CVer the first features are created when the model is run once to initialize GPUs for consistent speeds in later images:
https://github.com/ultralytics/yolov5/blob/33202b7f0bbfcc55860e8c0cafbf13f227e77a84/detect.py#L90-L92


We've made visualizing YOLOv5 🚀 architectures super easy. There are two main ways:

## `model.yaml`

Each model has a corresponding yaml file that displays the model architecture. Here is YOLOv5s, defined by `yolov5s.yaml`:
https://github.com/ultralytics/yolov5/blob/831773f5a23926658ee76459ce37550643432123/models/yolov5s.yaml#L12-L48

## TensorBoard Graph

Simply start training a model, and then view the TensorBoard Graph for an interactive view of the model architecture. This example shows YOLOv5s viewed in our [Notebook](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb) –  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

```python
# Tensorboard
%load_ext tensorboard
%tensorboard --logdir runs/train

# Train YOLOv5s on COCO128 for 3 epochs
!python train.py --weights yolov5s.pt --epochs 3
```

<img width="893" alt="Screenshot 2021-04-11 at 01 10 09" src="https://user-images.githubusercontent.com/26833433/114286928-349bd600-9a63-11eb-941f-7139ee6cd602.png">

**Comment by glenn-jocher** [comment]
@Zengyf-CVer good news 😃! Your original issue may now be fixed ✅ in PR #3920. This is a complete revamp of feature visualization with many fixes and improvements.

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
